src/cpp/rtps/builtin/discovery/endpoint/EDPServerListeners.cpp

```diff
@@ -13,171 +13,234 @@
 // limitations under the License.
 
 /**
- * @file EDPServerListener.cpp
+ * @file EDPServerListener2.cpp
  *
  */
 
-#include <rtps/builtin/discovery/endpoint/EDPServerListeners.h>
-#include <fastdds/rtps/builtin/discovery/endpoint/EDPServer.h>
-#include <fastdds/rtps/builtin/discovery/participant/PDPServer.h>
-
-#include <fastdds/rtps/builtin/data/WriterProxyData.h>
-#include <fastdds/rtps/builtin/data/ReaderProxyData.h>
-#include <fastrtps_deprecated/participant/ParticipantImpl.h>
-#include <fastdds/rtps/reader/StatefulReader.h>
 #include <fastdds/rtps/writer/StatefulWriter.h>
-
-#include <fastdds/rtps/history/ReaderHistory.h>
 #include <fastdds/rtps/history/WriterHistory.h>
+#include <fastdds/rtps/history/ReaderHistory.h>
+#include <fastdds/rtps/reader/RTPSReader.h>
 
-#include <fastdds/rtps/builtin/data/ParticipantProxyData.h>
-
-#include <mutex>
+#include <fastrtps/utils/fixed_size_string.hpp>
 
 #include <fastdds/dds/log/Log.hpp>
 
+#include "./EDPServerListeners.hpp"
+#include "./EDPServer.hpp"
+#include "../participant/PDPServer.hpp"
+
 namespace eprosima {
-namespace fastrtps {
+namespace fastdds {
 namespace rtps {
 
-EDPServerPUBListener::EDPServerPUBListener(
+using namespace eprosima::fastrtps::rtps;
+
+PDPServer* EDPServerPUBListener2::get_pdp()
```

Comment:
Careful, this should not be 2 anymore. Applies to entire class
```suggestion
PDPServer* EDPServerPUBListener::get_pdp()
```